sdk/src/test/java/com/qonversion/android/sdk/internal/user/controller/UserControllerTest.kt

```diff
@@ -305,4 +318,187 @@ internal class UserControllerTest {
             verify(exactly = 1) { mockLogger.error("Failed to update user cache", exception) }
         }
     }
+
+    @Nested
+    inner class OnAppForegroundTest {
+
+        @Test
+        fun `not the first foreground`() {
+            // given
+
+            // when
+            userController.onAppForeground(false)
+
+            // then
+            verify {
+                mockUserCacher wasNot called
+                mockUserService wasNot called
```

Comment:
Let's also check that `mockLogger`, `mockDataStorage`, `handleNewUserInfo()`  weren't called